husarnet/ports/unix/port_interface.cpp

```diff
@@ -236,18 +236,23 @@ namespace Port {
     fsync(fileno(f));
     fclose(f);
 
-    if(rename((path + ".tmp").c_str(), path.c_str()) < 0) {
-      LOG("could not write to %s (rename failed)", path.c_str());
-      return false;
-    }
-    return true;
+    return renameFile(path + ".tmp", path);
```

Comment:
:thumbsup: for updating the old code with newly added features and reducing duplication!